gcc 11.3 fails when compiling ecjpake drivers

### Summary
Command: `all.sh test_psa_crypto_config_accel_ecjpake_use_psa` fails on gcc 11.3 with commit 530da9 from development.
Output:
```
src/drivers/test_driver_pake.c: In function 'mbedtls_test_transparent_pake_output':
src/drivers/test_driver_pake.c:97:28: error: implicit conversion from 'psa_crypto_driver_pake_step_t' {aka 'enum psa_crypto_driver_pake_step'} to 'libtestdriver1_psa_crypto_driver_pake_step_t' {aka 'enum libtestdriver1_psa_crypto_driver_pake_step'} [-Werror=enum-conversion]
   97 |                 operation, step, output, output_size, output_length);
      |                            ^~~~
src/drivers/test_driver_pake.c: In function 'mbedtls_test_transparent_pake_input':
src/drivers/test_driver_pake.c:132:28: error: implicit conversion from 'psa_crypto_driver_pake_step_t' {aka 'enum psa_crypto_driver_pake_step'} to 'libtestdriver1_psa_crypto_driver_pake_step_t' {aka 'enum libtestdriver1_psa_crypto_driver_pake_step'} [-Werror=enum-conversion]
  132 |                 operation, step, input, input_length);
      |                            ^~~~
```
`-Werror=enum-conversion` is not present in gcc 9.5, but is present in gcc 10.4.

### System information

Mbed TLS version (number or commit id): 530da90081c5
Operating system and version: lubuntu 22.04: 5.15.0-60-generic #66-Ubuntu SMP Fri Jan 20 14:29:49 UTC 2023
Gcc version: 11.3.0
Command: `all.sh test_psa_crypto_config_accel_ecjpake_use_psa`

--- a/tests/src/drivers/test_driver_pake.c
+++ b/tests/src/drivers/test_driver_pake.c
@@ -94,7 +94,8 @@ psa_status_t mbedtls_test_transparent_pake_output(
         defined(LIBTESTDRIVER1_MBEDTLS_PSA_BUILTIN_PAKE)
         mbedtls_test_driver_pake_hooks.driver_status =
             libtestdriver1_mbedtls_psa_pake_output(
-                operation, step, output, output_size, output_length);
+                operation, (libtestdriver1_psa_crypto_driver_pake_step_t) step,
+                output, output_size, output_length);
 #elif defined(MBEDTLS_PSA_BUILTIN_PAKE)
         mbedtls_test_driver_pake_hooks.driver_status =
             mbedtls_psa_pake_output(
@@ -129,7 +130,8 @@ psa_status_t mbedtls_test_transparent_pake_input(
         defined(LIBTESTDRIVER1_MBEDTLS_PSA_BUILTIN_PAKE)
         mbedtls_test_driver_pake_hooks.driver_status =
             libtestdriver1_mbedtls_psa_pake_input(
-                operation, step, input, input_length);
+                operation, (libtestdriver1_psa_crypto_driver_pake_step_t) step,
+                input, input_length);
 #elif defined(MBEDTLS_PSA_BUILTIN_PAKE)
         mbedtls_test_driver_pake_hooks.driver_status =
             mbedtls_psa_pake_input(

